File: F#/F#/deque.fs
module FSharp.Deque

type Deque<'T>() =
    let data = ResizeArray<'T>()
    
    member this.addFront item = 
        data.Insert(0, item)
    
    member this.addRear item =
        data.Add item
    
    member this.removeRear () = 
        let res = data.[data.Count - 1]
        data.RemoveAt(data.Count - 1)
        res

    member this.removeFront () =
        let res = data.[0]
        data.RemoveAt(0)
        res
    
    member this.size () = data.Count


let palchecker word =
    let deque = Deque()
    word |> Seq.iter deque.addRear

    let rec loop () =
        if deque.size() > 1 then
            if deque.removeFront() <> deque.removeRear() then
                false
            else loop ()
        else true

    loop ()

File: Go/Deque/deque.go
package Dequeue
type Deque struct {
    data []interface{}
}

func (d *Deque) size() int {
    return len(d.data)
}

func (d *Deque) addFront(item interface{}) {
    d.data = append([]interface{}{item}, d.data...)
}

func (d *Deque) addRear(item interface{}) {
    d.data = append(d.data, item)
}

func (d *Deque) removeRear() interface{} {
    res := d.data[len(d.data) - 1]
    d.data = d.data[:len(d.data) - 1]
    return res
}

func (d *Deque) removeFront() interface{} {
    res := d.data[0]
    d.data = d.data[1:]
    return res
}

func palchecker(word string) bool {
    deque := new(Deque)
    for _, c := range word {
        deque.addRear(c)
    }

    for deque.size() > 1 {
        if deque.removeFront() != deque.removeRear() {
            return false
        }
    }
    return true
}